packages/scripts/src/upgrade-tests/162/1_Propose.ts

```diff
@@ -1,85 +1,344 @@
-// Then execute the script from core:
-// OPTIMISTC_ORACLE_V2=<OPTIMISTC-ORACLE-V2-ADDRESS> yarn hardhat run ./src/upgrade-tests/162/1_Propose.ts --network localhost
+// Then execute the script:
+// OPTIMISTIC_ORACLE_V2_10=<OPTIMISM-OOV2-ADDRESS> \
+// NODE_URL_10=<OPTIMISM-NODE-URL> \
+// \
+// OPTIMISTIC_ORACLE_V2_288=<BOBA-OOV2-ADDRESS> \
+// NODE_URL_288=<OPTIMISM-NODE-URL> \
+// \
+// OPTIMISTIC_ORACLE_V2_137=<POLYGON-OOV2-ADDRESS> \
+// NODE_URL_137=<OPTIMISM-NODE-URL> \
+// \
+// OPTIMISTIC_ORACLE_V2_42161=<ARBITRUM-OOV2-ADDRESS> \
+// NODE_URL_42161=<OPTIMISM-NODE-URL> \
+// \
+// OPTIMISTC_ORACLE_V2=<MAINNET-OOV2-ADDRESS> \
+// \
+// yarn hardhat run ./src/upgrade-tests/162/1_Propose.ts  --network localhost
 
 const hre = require("hardhat");
+import * as path from "path";
+import fs from "fs";
 
 const { RegistryRolesEnum } = require("@uma/common");
-
-import { Finder, Governor, Proposer, Registry } from "@uma/contracts-node/typechain/core/ethers";
-import { getContractInstance } from "../../utils/contracts";
+import { BigNumberish } from "@ethersproject/bignumber";
+import { BytesLike } from "@ethersproject/bytes";
+import {
+  ArbitrumParentMessenger,
+  Finder,
+  Governor,
+  GovernorHub,
+  GovernorRootTunnel,
+  Proposer,
+  Registry,
+} from "@uma/contracts-node/typechain/core/ethers";
+import { BaseContract, PopulatedTransaction, Signer } from "ethers";
+import { getContractInstance, getContractInstanceByUrl } from "../../utils/contracts";
 
 // PARAMETERS
 const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
 const deployed_optimistic_oracle_address = process.env["OPTIMISTC_ORACLE_V2"];
 
 const OPTIMISTIC_ORACLE_V2 = "OptimisticOracleV2"; // TODO use interfaceName.OptimisticOracle
 
-async function main() {
-  if (!deployed_optimistic_oracle_address) throw new Error("OPTIMISTC_ORACLE_V2 environment variable not set");
+// CONSTANTS
+const POLYGON_ID = 137;
+const ARBITRUM_ID = 42161;
+
+// Env vars
+const NODE_URL_ENV = "NODE_URL_";
+const OPTIMISTIC_ORACLE_V2_ENV = "OPTIMISTIC_ORACLE_V2_";
+
+export interface RelayProposal {
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+  chainId: BigNumberish;
+}
+export interface RelayRecords {
+  block: number;
+  governorRootTunnel: RelayProposal[];
+  governorHub: RelayProposal[];
+}
+
+const relayGovernanceRootTunnelMessage = async (
+  targetAddress: string,
+  tx: PopulatedTransaction,
+  governorRootTunnel: GovernorRootTunnel
+): Promise<{
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}> => {
+  if (!tx.data) throw new Error("Transaction has no data");
+  const relayGovernanceData = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  console.log("- relayGovernanceData", relayGovernanceData);
+  const relay = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  const relayMessage = relay.data;
```

Comment:
This is much easier than using the ethers Interface classes. Didn't realize you could generate transaction data this way.